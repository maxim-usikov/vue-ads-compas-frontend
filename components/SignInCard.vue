<template>
  <VCard class="elevation-12">
    <VToolbar dark color="primary">
      <VToolbarTitle>Sign in</VToolbarTitle>
    </VToolbar>

    <VCardText>
      <VAlert :value="error" type="error" class="mb-4">
        {{ error.message }}
      </VAlert>

      <VForm v-model="valid">
        <VTextField
          v-model="username"
          :rules="rules.username"
          prepend-icon="person"
          name="username"
          label="Email"
          type="text"
          required
          @input="error = false"
        />

        <VTextField
          v-model="password"
          :rules="rules.password"
          prepend-icon="lock"
          name="password"
          label="Password"
          type="password"
          required
          @input="error = false"
        />
      </VForm>
    </VCardText>

    <VCardActions>
      <VSpacer />

      <VBtn
        color="primary"
        :disabled="!valid || loading"
        :loading="loading"
        @click="handleSignIn"
        >Sign in</VBtn
      >
    </VCardActions>
  </VCard>
</template>

<script>
export default {
  name: 'SignInCard',

  data() {
    return {
      loading: false,
      valid: false,
      error: false,
      username: '',
      password: '',
      rules: {
        username: [v => !!v || 'E-mail is required'],
        password: [v => !!v || 'Password is required']
      }
    }
  },

  methods: {
    async handleSignIn() {
      const { username, password } = this

      try {
        this.loading = true
        this.error = false

        await this.$auth.loginWith('GrantPassword', {
          data: {
            username,
            password
          }
        })
      } catch (err) {
        if (err.response && err.response.data) {
          this.error = err.response.data
        }
      } finally {
        this.loading = false
      }
    }
  }
}
</script>
