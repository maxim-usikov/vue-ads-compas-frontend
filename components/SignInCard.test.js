import { shallowMount } from '@vue/test-utils'

import SignInCard from './SignInCard.vue'

describe('SignInCard', () => {
  let $c

  beforeEach(() => {
    $c = shallowMount(SignInCard)
  })

  it('is a Vue instance', () => {
    expect($c.isVueInstance()).toBeTruthy()
  })

  it('name is `SignInCard`', () => {
    expect($c.name()).toBe('SignInCard')
  })

  it('renders correctly', () => {
    expect($c.element).toMatchSnapshot()
  })
})
