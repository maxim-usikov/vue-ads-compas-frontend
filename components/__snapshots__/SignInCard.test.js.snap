// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInCard renders correctly 1`] = `
<vcard-stub
  class="elevation-12"
  tag="div"
>
  <vtoolbar-stub
    color="primary"
    dark="true"
    scrollthreshold="300"
  >
    <vtoolbartitle-stub>
      Sign in
    </vtoolbartitle-stub>
  </vtoolbar-stub>
   
  <vcardtext-stub>
    <valert-stub
      class="mb-4"
      type="error"
    >
      
      
    
    </valert-stub>
     
    <vform-stub>
      <vtextfield-stub
        label="Email"
        name="username"
        prepend-icon="person"
        required=""
        rules="v => !!v || 'E-mail is required'"
        type="text"
        value=""
      />
       
      <vtextfield-stub
        label="Password"
        name="password"
        prepend-icon="lock"
        required=""
        rules="v => !!v || 'Password is required'"
        type="password"
        value=""
      />
    </vform-stub>
  </vcardtext-stub>
   
  <vcardactions-stub>
    <vspacer-stub />
     
    <vbtn-stub
      activeclass="v-btn--active"
      color="primary"
      disabled="true"
      tag="button"
      type="button"
    >
      Sign in
    </vbtn-stub>
  </vcardactions-stub>
</vcard-stub>
`;
